### PR TITLE
fix: use @types/node buffer type rather than lib.dom.ts

### PR DESCRIPTION
**Pull request by jeswr**
This PR fixes #<issue ID>.

As discussed in https://github.com/microsoft/TypeScript/issues/53668
the @types/node definition of a buffer is looser and hence we should
use that in order to be compaible with web buffer types and node
buffer types

- [x] I've added a unit test to test for potential regressions of this bug.
- [x] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/resource/file.test.ts [hunk 1]
```diff
@@ -376,339 +377,380 @@ describe("Write non-RDF data into a folder", () => {
...
+          fetch: jest.Mocked<typeof window.fetch>;
+        };
+
+        fetcher.fetch = setMockOnFetch(fetcher.fetch);
+
+        const mockTextBlob = new Blob(["mock blob data"]);
+        const savedFile = await saveFileInContainer(
+          "https://some.url",
+          mockTextBlob
+        );
```
Comment (NSeydoux) [inline src/resource/file.test.ts:524]: Should the default `mockBlob` have no `type`, and this be replaced with:
```suggestion
        const savedFile = await saveFileInContainer(
          "https://some.url",
          data
        );
```
?

If so, we should have a separate test for the `instanceof` check, to make sure the generic typing works properly.

File: src/resource/file.test.ts [hunk 2]
```diff
@@ -376,339 +377,380 @@ describe("Write non-RDF data into a folder", () => {
...
+  const mockBuffer = Buffer.from("mock blob data");
+  const mockNodeBuffer = NodeBuffer.from("mock blob data");
+
+  describe.each([
+    ["blob", mockBlob],
+    ["buffer", mockBuffer],
+    ["nodeBuffer", mockNodeBuffer],
+  ] as [string, Blob | Buffer][])(
+    "blob, buffer and nodeBuffer test",
+    (_, data) => {
```
Comment (NSeydoux) [inline src/resource/file.test.ts:611]: ditto

File: e2e/node/resource.test.ts [hunk 3]
```diff
@@ -151,6 +152,49 @@ describe("Authenticated end-to-end", () => {
...
+        new Blob(["test"]),
+        fetchOptions
+      );
+      const sessionDataset = await getSolidDataset(
+        sessionResource,
+        fetchOptions
+      );
+
+      // Eslint isn't detecting the fact that this is inside an it statement
+      // for an unknown reason
```
Comment (NSeydoux) [inline e2e/node/resource.test.ts:188]:
```suggestion
      // because of the conditional.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -16,6 +16,13 @@ The following changes are pending, and will be applied on the next major release
 
 The following changes have been implemented but not released yet:
 
+### Bugfixes
+
+- `Buffer` type: As discussed in microsoft/TypeScript#53668 the
+  @types/node definition of a buffer is looser and hence we now
```
Comment (NSeydoux) [inline CHANGELOG.md:22]:
```suggestion
  @types/node definition of a buffer is looser than the DOM one (the latter being TS' default), and hence we now
```

**Comment by jeswr** [comment]
Do not merge until https://github.com/inrupt/solid-client-js/pull/1982 is released

File: src/resource/file.test.ts [hunk 5]
```diff
@@ -376,81 +377,164 @@ describe("Write non-RDF data into a folder", () => {
...
-      "Content-Type": "binary",
-      Slug: "someFileName",
-    });
-    expect(mockCall[1]?.body).toEqual(mockBlob);
-  });
+  describe.each([
+    ["blob", mockBlob],
+    ["buffer", mockBuffer],
+    ["nodeBuffer", mockNodeBuffer],
+  ] as [string, Blob | Buffer][])(
```
Comment (NSeydoux) [inline src/resource/file.test.ts:384]: I don't think the type assertion is necessary here

**Review by NSeydoux — APPROVED** [review]

**Pull request merged**